java/vector/src/main/java/org/apache/arrow/vector/dictionary/Dictionary.java

```diff
@@ -63,7 +64,7 @@ public boolean equals(Object o) {
       return false;
     }
     Dictionary that = (Dictionary) o;
-    return Objects.equals(encoding, that.encoding) && Objects.equals(dictionary, that.dictionary);
+    return Objects.equals(encoding, that.encoding) && Validator.compareFieldVectors(dictionary, that.dictionary);
```

Comment:
It would be better to compare encodings with the Validator also, e.g. a function `Validator.compareDictionaryEncodings`. Isn't `Object.equals` incorrect here too?

Reply:
I think Object.equals works fine with DictionaryEncoding since it implemented hashCode & equals already.

Reply:
Do you mean DictionaryEncoding? I think Object.equals will call DictionaryEncoding#equals?

Reply:
My mistake, you are correct